src/Common/src/Interop/OSX/Interop.RunLoop.cs

```diff
@@ -26,7 +27,11 @@ internal static partial class RunLoop
         /// <summary>
         /// Starts the current thread's RunLoop. If the RunLoop is already running, creates a new, nested, RunLoop in the same stack.
         /// </summary>
+#if MONO
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+#else
         [DllImport(Interop.Libraries.CoreFoundationLibrary)]
+#endif
```

Comment:
In the process of integrating the FSEvent watcher I only needed the one above change in corefx. I know the ifdefs are more extensive in other parts that are being ported, as the Mono build deals with platform-specific interop in a different way.

@luhenry @marek-safar ^^